reVX/wind_dirs/prominent_wind_dirs.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+"""
+Aggregate powerrose and sort directions by dominance
+"""
+import h5py
+import numpy as np
+import pandas as pd
+
+from reV.supply_curve.aggregation import Aggregation
+from reV.supply_curve.points import SupplyCurveExtent
+
+
+class ProminentWindDirections(Aggregation):
+    """
+    Aggregate PowerRose to Supply Curve points and sort directions in order
+    of prominence. Then convert to equivalent sc_point_gid
+    """
+    def __init__(self, power_rose_h5_fpath, excl_fpath,
+                 agg_dset='powerrose_100m', tm_dset='techmap_wtk',
+                 resolution=64):
+        """
+        Parameters
+        ----------
+        power_rose_h5_fpath : str
+            Filepath to .h5 file containing powerrose data
+        excl_fpath : str
+            Filepath to exclusions h5 with techmap dataset.
+        agg_dset : str, optional
+            Powerrose dataset to aggreate, by default 'powerrose_100m'
+        tm_dset : str, optional
+            Dataset name in the techmap file containing the
+            exclusions-to-resource mapping data,
+            by default 'techmap_wtk'
+        resolution : int, optional
+            SC resolution, must be input in combination with gid. Prefered
+            option is to use the row/col slices to define the SC point instead,
+            by default 64
+        """
+        super().__init__(excl_fpath, power_rose_h5_fpath, tm_dset, agg_dset,
+                         resolution=resolution)
+
+    @staticmethod
+    def _map_direction_pos(power_rose_h5_fpath):
+        """
+        Map powerrose directions to sc row and column shifts
+
+        Parameters
+        ----------
+        power_rose_h5_fpath : str
+            Filepath to .h5 file containing powerrose
+
+        Returns
+        -------
+        list
+            Pos of major cardinal directions in power rose data
+        """
+        directions = ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
+
+        with h5py.File(power_rose_h5_fpath, 'r') as f:
+            cardinal_dirs = list(f['cardinal_directions'][...].astype(str))
+
+        dir_pos = [cardinal_dirs.index(d) for d in directions]
+
+        return dir_pos
+
+    @staticmethod
+    def _compute_neighbors(sc_point_gids, points):
+        """
+        Compute neighboring supply curve point gids in following order:
+        ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
+
+        Parameters
+        ----------
+        sc_point_gids : list | ndarray
+            Supply curve point gids to get neighbors for
+        points : pandas.DataFrame
+            Supply curve point gid to row, col index mapping
+
+        Returns
+        -------
+        neighbor_gids : ndarray
+            Neighboring supply curve point gids
+        """
+        sc_points = points.loc[sc_point_gids]
+        rows, cols = sc_points.values.T
+        shape = points['row_ind'].max(), points['col_ind'].max()
+
+        row_shifts = [-1, -1, 0, 1, 1, 1, 0, -1]
```

Comment:
Can you explain this to me?

Reply:
Which part? I replaced lines 84 -> 86 with a new method that takes the sc_point_gids and shape and computes the row and col indices.

The rest of the method computes the gids from the row and column indices for all of the neighbors